lisa/analysis/rta.py

```diff
@@ -0,0 +1,245 @@
+    def from_task_names(cls, task_names, log_dir):
+        """
+        Build a :class:`RTALogAnalysis` from a list of task names
+
+        :param task_names: List of task names to look for
+        :type task_names: list(str)
+
+        :param log_dir: Folder containing RTApp log files
+        :type log_dir: str
+        """
+        def find_log_file(task_name, log_dir):
+            log_file = os.path.join(log_dir, cls.RTA_LOG_PATTERN.format(task_name))
+            if not os.path.isfile(log_file):
+                raise ValueError('No rt-app logfile found for task [{}]'.format(
+                    task_name
+                ))
+            return log_file
+
+        task_log_map = {
+            task_name: find_log_file(task, log_dir)
+            for task_name in tasks
+        }
+        return cls(task_log_map)
+
+    @staticmethod
+    def _parse_df(logfile):
+        df = pd.read_csv(logfile,
+                sep='\s+',
+                skiprows=1,
+                header=0,
+                usecols=[1,2,3,4,7,8,9,10],
+                names=[
+                    'Cycles', 'Run' ,'Period', 'Timestamp',
+                    'Slack', 'CRun', 'CPeriod', 'WKPLatency'
+                ])
+        # Normalize time to [s] with origin on the first event
+        start_time = df['Timestamp'][0]/1e6
+        df['Time'] = df['Timestamp']/1e6 - start_time
+        df.set_index(['Time'], inplace=True)
+        # Add performance metrics column, performance is defined as:
+        #             slack
+        #   perf = -------------
+        #          period - run
+        df['PerfIndex'] = df['Slack'] / (df['CPeriod'] - df['CRun'])
+
+        return df
+
+    @property
+    def tasks(self):
+        """
+        List of tasks for which performance data have been loaded
+        """
+        return sorted(self.perf_data.keys())
+
+    def get_log_file(self, task):
+        """
+        Return the logfile for the specified task
+
+        :param task: Name of the task that we want the logfile of.
+        :type task: str
+        """
+        return self.perf_data[task]['logfile']
+
+    def get_df(self, task):
+        """
+        Return the pandas dataframe with the performance data for the
+        specified task
+
+        :param task: Name of the task that we want the performance dataframe of.
+        :type task: str
+        """
+        return self.perf_data[task]['df']
+
+    def save_plot(self, figure, filepath, img_format="png"):
+        """
+        .. see: :class:`lisa.analysis.base.AnalysisHelper.save_plot`
+
+        .. note::
+            We force specifying the filepath, since the
+            :class:`lisa.analysis.base.AnalysisHelper` would use a `trace`
+            attribute that does not exists otherwise.
+        """
+        return super().save_plot(
+            figure=figure,
+            filepath=filepath,
+            img_format=img_format
+        )
+
+    def plot_perf(self, task, filepath=None, axis=None):
```

Comment:
On this simple example:

```python
from lisa.analysis.rta import RTALogAnalysis
rta = RTALogAnalysis.from_dir('/data/Code/lisa/results/20180209_UtilEst_JunoR2/')
rta.tasks
rta.plot_perf('jumper')
```

I get this error:

```python
AttributeError                            Traceback (most recent call last)
<ipython-input-6-9d6013597ea8> in <module>
----> 1 rta.plot_perf('jumper')

/data/Code/lisa2/lisa/analysis/rta.py in plot_perf(self, task, filepath, axis)
    202             axis.set_ylim(0, 2)
    203 
--> 204         return self._do_plot(plotter, filepath, axis)
    205 
    206     def plot_latency(self, task, filepath=None, axis=None):

/data/Code/lisa2/lisa/analysis/base.py in _do_plot(self, plotter, filepath, axis)
    154 
    155         if local_fig:
--> 156             self.save_plot(fig, filepath)
    157         return axis
    158 

/data/Code/lisa2/lisa/analysis/rta.py in save_plot(self, figure, filepath, img_format)
    183             figure=figure,
    184             filepath=filepath,
--> 185             img_format=img_format
    186         )
    187 

/data/Code/lisa2/lisa/analysis/base.py in save_plot(self, figure, filepath, img_format)
    138             caller = inspect.stack()[1][3]
    139             filepath = os.path.join(
--> 140                 self.trace.plots_dir,
    141                 "{}.{}.{}".format(module, caller, img_format))
    142 

AttributeError: 'RTALogAnalysis' object has no attribute 'trace'
```

which also breaks loops like:

```python
for task in rta.tasks:
    rta.plot_latency(task)
```

Reply:
Same warnings for the following two methods